tests/drivers/dma/chan_blen_transfer/src/test_dma.c

```diff
@@ -26,7 +26,9 @@
 #define DMA_DEVICE_NAME CONFIG_DMA_0_NAME
 #define RX_BUFF_SIZE (48)
 
+__aligned(4)
```

Comment:
aren't {dest,source}_data_size 1, so why the alignment need?

I'm guessing we need to figure out a broader solution here.

Reply:
Ah, I put this in so I could experiment with different transfer data sizes.  The test passes as-is, but if you change {source,dest}_data_size to, say, 4, then the buffer isn't aligned and the test fails.

I'll split this out into a different PR and we can discuss there.